packages/sirius-web/backend/sirius-web/src/test/java/org/eclipse/sirius/web/application/controllers/projects/ProjectDownloadControllerIntegrationTests.java

```diff
@@ -0,0 +1,189 @@
+        } catch (IOException exception) {
+            fail(exception.getMessage());
+        }
+    }
+
+    @Test
+    @DisplayName("Given a project, when the download of the project is requested, then the representation data are retrieved")
+    @Sql(scripts = { "/scripts/initialize.sql" }, executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = { "/scripts/cleanup.sql" }, executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD, config = @SqlConfig(transactionMode = SqlConfig.TransactionMode.ISOLATED))
+    public void givenProjectWhenTheDownloadOfProjectIsRequestedThenTheRepresentationDataAreRetrieved() {
+        var response = this.download(TestIdentifiers.ECORE_SAMPLE_PROJECT);
+
+        try (var inputStream = new ZipInputStream(response.getBody().getInputStream())) {
+            var zipEntries = this.toZipEntries(inputStream);
+            String representationPath = "Ecore Sample/representations/" + TestIdentifiers.EPACKAGE_PORTAL_REPRESENTATION.toString() + ".json";
+            assertThat(zipEntries).isNotEmpty().containsKey(representationPath);
+
+            String representationContentExpected = this.getExpectedRepresentation();
+            String representationContent = new String(zipEntries.get(representationPath).toByteArray(), StandardCharsets.UTF_8);
+            assertEquals(representationContentExpected, representationContent);
+        } catch (IOException exception) {
+            fail(exception.getMessage());
+        }
+    }
+
+    private ResponseEntity<Resource> download(UUID projectId) {
+        this.givenCommittedTransaction.commit();
+
+        var uri = "http://localhost:" + this.port + "/api/projects/" + projectId.toString();
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.setAccept(List.of(MediaType.parseMediaType("application/zip")));
+        HttpEntity<String> entity = new HttpEntity<>(null, headers);
+
+        var response = new TestRestTemplate().exchange(uri, HttpMethod.GET, entity, Resource.class);
+        assertThat(response.getStatusCode()).isEqualTo(HttpStatus.OK);
+
+        return response;
+    }
+
+    private HashMap<String, ByteArrayOutputStream> toZipEntries(ZipInputStream inputStream) {
+        HashMap<String, ByteArrayOutputStream> zipEntries = new HashMap<>();
+
+        try {
+            var zipEntry = inputStream.getNextEntry();
+            while (zipEntry != null) {
+                if (!zipEntry.isDirectory()) {
+                    String name = zipEntry.getName();
+                    ByteArrayOutputStream entryBaos = new ByteArrayOutputStream();
+                    inputStream.transferTo(entryBaos);
+                    zipEntries.put(name, entryBaos);
+                }
+                zipEntry = inputStream.getNextEntry();
+            }
+        } catch (IOException exception) {
+            fail(exception.getMessage());
+        }
+
+        return zipEntries;
+    }
+
```

Comment:
For all the code below and other tests, we are using Java17 so please use multiline strings and don't escape double quotes characters

Reply:
Done